astropy/units/physical.py

```diff
@@ -1,133 +1,473 @@
+        return "{" + str(self._physical_type_list)[1:-1] + "}"
+
+    def __repr__(self):
+        if len(self._physical_type) == 1:
+            names = "'" + self._physical_type_list[0] + "'"
+        else:
+            names = self._name_string_as_ordered_set
+        return f"PhysicalType({names})"
+
+    def __str__(self):
+        if len(self._physical_type) == 1:
+            return self._physical_type_list[0]
+        else:
+            return self._name_string_as_ordered_set
+
+    @staticmethod
+    def _dimensionally_compatible_unit(obj):
+        """
+        If a unit is passed in, return that unit.  If a physical type is
+        passed in, return a unit that corresponds to that physical type.
+        If a real number is passed in, return a dimensionless unit.
+        Otherwise, return `NotImplemented`.
+        """
+        if isinstance(obj, core.UnitBase):
+            return _replace_temperatures_with_kelvin(obj)
+        elif isinstance(obj, PhysicalType):
+            return obj._unit
+        elif isinstance(obj, numbers.Real):
+            return core.dimensionless_unscaled
+        else:
+            return NotImplemented
+
+    def _dimensional_analysis(self, other, operation):
+        other_unit = self._dimensionally_compatible_unit(other)
+        if other_unit is NotImplemented:
+            return NotImplemented
+        other_unit = _replace_temperatures_with_kelvin(other_unit)
+        new_unit = getattr(self._unit, operation)(other_unit)
+        return new_unit.physical_type
+
+    def __mul__(self, other):
+        return self._dimensional_analysis(other, "__mul__")
+
+    def __rmul__(self, other):
+        return self.__mul__(other)
+
+    def __truediv__(self, other):
+        return self._dimensional_analysis(other, "__truediv__")
+
+    def __rtruediv__(self, other):
+        other = self._dimensionally_compatible_unit(other)
+        if other is NotImplemented:
+            return NotImplemented
+        return other.physical_type._dimensional_analysis(self, "__truediv__")
+
+    def __pow__(self, power):
+        return (self._unit ** power).physical_type
+
+    def __hash__(self):
+        return hash(self._physical_type_id)
+
+    def __len__(self):
+        return len(self._physical_type)
+
+    # We need to prevent operations like where a Unit instance left
+    # multiplies a PhysicalType instance from returning a `Quantity`
+    # instance with a PhysicalType as the value.  We can do this by
+    # preventing np.array from casting a PhysicalType instance as
+    # an object array.
+    __array__ = None
+
+
+def _get_names_in_use_except_from(unit):
+    """
+    Get a `set` containing all of the physical type names in use, except
+    for the names that correspond to ``unit``.
+    """
+    all_names_in_use = set(_unit_physical_mapping.keys())
+    id = unit._get_physical_type_id()
+    if id in _physical_unit_mapping.keys():
+        names_for_this_unit = set(_physical_unit_mapping[id])
```

Comment:
Just `return all_names_in_use - set(...)` here and then `return all_names_in_use` in the `else` clause.

Or, maybe better, call it `names_in_use` and here do `names_in_use -= set(_physical_unit_mapping[id])`